DropboxSync.BLL/Services/FileService.cs

```diff
@@ -0,0 +1,193 @@
+        public FileService(ILogger<FileService> logger)
+        {
+            _logger = logger ??
+                throw new ArgumentNullException(nameof(logger));
+            _httpClient = new HttpClient();
+
+            string appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+
+            string filesDirectory = Path.Join(appData, FILE_DOWNLOAD_DIR);
+
+            if (!Directory.Exists(filesDirectory))
+            {
+                Directory.CreateDirectory(filesDirectory);
+            }
+        }
+
+        /// <summary>
+        /// Download the file from API and save it locally. Finally returns an <see cref="SavedFile"/> object containing needed data
+        /// </summary>
+        /// <param name="fileId">The ID of the file to download</param>
+        /// <returns><see cref="SavedFile"/> object if download and save was successfull. <c>null</c> Otherwise</returns>
+        public async Task<SavedFile?> DownloadFile(string fileId)
+        {
+            if (string.IsNullOrEmpty(fileId)) throw new ArgumentNullException(nameof(fileId));
+
+            SavedFile? finalOutput = null;
+
+            if (!await GetToken())
+            {
+                _logger.LogError("{date} | Could not retrieve Access Token to the API.", DateTime.Now);
+                return finalOutput;
+            }
+
+            string fileDownloadUrl = $"{API_BACKEND_URL}/api/resource/download?id={fileId}";
+
+            HttpResponseMessage response = await _httpClient.GetAsync(fileDownloadUrl);
+
+            if (response is null)
+            {
+                _logger.LogError("{date} | Something went wrong on api call to route \"{fileDownloadUrl}\"",
+                    DateTime.Now, fileDownloadUrl);
+                return finalOutput;
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.LogInformation("{date} | The API call was unsuccesfull. Reponse status is \"{responseCode}\"",
+                    DateTime.Now, response.StatusCode);
+
+                return finalOutput;
+            }
+
+            if (response.Content is null)
+            {
+                _logger.LogError("{date} | Response content is null!", DateTime.Now);
+                return finalOutput;
+            }
+
+
+            byte[] fileData = await response.Content.ReadAsByteArrayAsync();
+
+            if (fileData is null)
+            {
+                _logger.LogError("{date} | Byte array from file is null", DateTime.Now);
+                return finalOutput;
+            }
+
+            string? fileName = response.Content.Headers?.ContentDisposition?.FileName;
+
+            if (string.IsNullOrEmpty(fileName))
+            {
+                _logger.LogError("{date} | The file name for file with ID \"{fileId}\" could not be retrieved!",
+                    DateTime.Now, fileId);
+                return finalOutput;
+            }
+
+            fileName = fileName.Replace("\"", "");
```

Comment:
As always, I wrote it for a reason and completly forgot why. I rather leave it, it doesn't break the code because whatever Os I use, no file name can contain a `/` or `\`. (Correct me if I'm wrong)  so this line would just do nothing

Reply:
yep, so I think that line can be removed. you can search for a method in .net's api to sanitize filename
 if you want to be sure the file name will be correct, otherwise this is dead code.